app/src/main/java/com/infomaniak/swisstransfer/ui/components/SwissTransferTopAppBar.kt

```diff
@@ -31,6 +31,40 @@ import com.infomaniak.swisstransfer.ui.images.icons.Cross
 import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
 import com.infomaniak.swisstransfer.ui.utils.PreviewAllWindows
 
+@Composable
+fun SwissTransferTopAppBar(
+    title: String? = null,
+    @StringRes titleRes: Int,
+    navigationMenu: TopAppBarButton? = null,
+    vararg actionMenus: TopAppBarButton,
+) {
+    if (title != null) {
+        SwissTransferTopAppBar(title = title, navigationMenu = navigationMenu, actionMenus = actionMenus)
+    } else {
+        SwissTransferTopAppBar(titleRes = titleRes, navigationMenu = navigationMenu, actionMenus = actionMenus)
+    }
+}
+
+@Composable
+@OptIn(ExperimentalMaterial3Api::class)
+fun SwissTransferTopAppBar(
+    title: String,
+    navigationMenu: TopAppBarButton? = null,
+    vararg actionMenus: TopAppBarButton,
+) {
+    TopAppBar(
+        colors = TopAppBarDefaults.topAppBarColors(
+            containerColor = SwissTransferTheme.materialColors.tertiary,
+            titleContentColor = SwissTransferTheme.colors.toolbarTextColor,
+            actionIconContentColor = SwissTransferTheme.colors.toolbarIconColor,
+            navigationIconContentColor = SwissTransferTheme.colors.toolbarIconColor,
+        ),
+        title = { Text(text = title, style = SwissTransferTheme.typography.h2) },
+        navigationIcon = { navigationMenu?.let { MenuButton(navigationMenu) } },
+        actions = { actionMenus.forEach { actionMenu -> MenuButton(actionMenu) } },
+    )
+}
+
```

Comment:
Don't duplicate the code like this, define the inner component as taking in a String and if given a stringRes in the outer component, call the inner component by reading the string ressource. This way both cases call the same bit of code. Always.

Always try to make it so the similar code like this (where both cases should always be the same) is defined depending on one another so nobody can ever introduce any form of difference between the two even if they tried